Sources/GRPC/ConnectionPool/ConnectionPool+Waiter.swift

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import NIO
+import NIOHTTP2
+
+extension ConnectionPool {
+  internal struct Waiter {
```

Comment:
I think this object is probably a `class`, not a `struct`. In particular it has identity, as manifested by the fact that we cancel it. Given that we're also closing over it in some cases, redesigning this as a `class` is likely better.